src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// Hosts a gRPC service (via <see cref="DashboardService"/>) that a dashboard can connect to.
+/// Configures DI and networking options for the service.
+/// </summary>
+internal sealed class DashboardServiceHost : IHostedService
+{
+    private const string DashboardServiceUrlVariableName = "DOTNET_DASHBOARD_GRPC_ENDPOINT_URL";
+    private const string DashboardServiceUrlDefaultValue = "http://localhost:18999";
+
+    /// <summary>
+    /// <see langword="null"/> if <see cref="DistributedApplicationOptions.DashboardEnabled"/> is <see langword="false"/>.
+    /// </summary>
+    private readonly WebApplication? _app;
+
+    public DashboardServiceHost(
+        DistributedApplicationOptions options,
+        DistributedApplicationModel applicationModel,
+        KubernetesService kubernetesService)
+    {
+        if (!options.DashboardEnabled)
+        {
+            return;
+        }
+
+        var builder = WebApplication.CreateBuilder();
+
+        builder.Services.AddGrpc();
+        builder.Services.AddSingleton(applicationModel);
+        builder.Services.AddSingleton(kubernetesService);
+        builder.Services.AddSingleton<DashboardServiceData>();
+
+        builder.WebHost.ConfigureKestrel(ConfigureKestrel);
+
+        _app = builder.Build();
+
+        _app.MapGrpcService<DashboardService>();
+        _app.MapGet("/", () => "Communication with gRPC endpoints must be made through a gRPC client. To learn how to create a client, visit: https://go.microsoft.com/fwlink/?linkid=2086909");
```

Comment:
This can be removed. No one should ever be browsing to this port. And if they do, they're probably get an error about HTTP/2 on an unsecured port before they see this error.

Reply:
Indeed. You'd see this, not the route:

![image](https://github.com/dotnet/aspire/assets/350947/77fa49b1-6325-44ae-86f2-017e17432cc0)